org.scala-ide.sdt.core/src/org/scalaide/ui/internal/actions/AbstractPopupAction.scala

```diff
@@ -1,20 +1,38 @@
 package org.scalaide.ui.internal.actions
 
+import org.eclipse.core.commands.AbstractHandler
+import org.eclipse.core.commands.ExecutionEvent
 import org.eclipse.core.resources.IProject
 import org.eclipse.core.runtime.IAdaptable
 import org.eclipse.jface.action.IAction
 import org.eclipse.jface.viewers.ISelection
 import org.eclipse.jface.viewers.IStructuredSelection
-import org.eclipse.core.runtime.Platform
 import org.eclipse.ui.IObjectActionDelegate
 import org.eclipse.ui.IWorkbenchPart
 import org.scalaide.util.eclipse.EclipseUtils.RichAdaptable
+import org.scalaide.util.eclipse.EditorUtils
 
-trait AbstractPopupAction extends IObjectActionDelegate {
+/**
+ * This traits contains definitions to handle the invocation of menu entries
+ * that are shown in the "Scala" context menu.
+ *
+ * Extends [[AbstractHandler]] to allow users to call this action by
+ * key bindings.
+ */
+trait AbstractPopupAction extends AbstractHandler with IObjectActionDelegate {
   private var selectionOption: Option[ISelection] = None
 
+  /**
+   * This method is called if either the menu entry is invoked or the handler
+   * (if it exists) is called when an [[IResource]] element is selected.
+   */
   def performAction(project: IProject)
 
+  override def execute(event: ExecutionEvent): AnyRef = {
+    EditorUtils.resourceOfActiveEditor flatMap (r ⇒ Option(r.getProject)) foreach performAction
```

Comment:
I'm not a fan of unicode characters like this one, since Eclipse can't handle them properly (line spacing increases suddenly in those files). But I can't stop progress, I guess....